includes/Context.php

```diff
@@ -373,7 +373,10 @@ public function get_amp_mode() {
 			) {
 				return self::AMP_MODE_SECONDARY;
 			}
-		} elseif ( function_exists( 'amp_is_canonical' ) ) {
+		} elseif (
+			function_exists( 'amp_is_canonical' ) ||
+			is_singular( 'web-story' )
+		) {
```

Comment:
This is too late for this condition since the first line in `get_amp_mode` will return early if the AMP plugin is not installed which is not a requirement/dependency of the Web Stories plugin.